pandas/io/common.py

```diff
@@ -201,6 +201,33 @@ def _validate_header_arg(header):
                         "header=int or list-like of ints to specify "
                         "the row(s) making up the column names")
 
+def _stringify_path(filepath_or_buffer):
+    """Return the argument coerced to a string if it was a pathlib.Path
+       or a py.path.local.LocalPath
+
+    Parameters
+    ----------
+    filepath_or_buffer : object to be converted
+
+    Returns
+    -------
+    str_filepath_or_buffer : a the string version of the input path
+    """
+    try:
+        import pathlib
+        if isinstance(filepath_or_buffer, pathlib.Path):
+            return text_type(filepath_or_buffer)
+    except:
+        pass
+    try:
+        from py.path.local import LocalPath
```

Comment:
imports are cached right? So there shouldn't be any performance downsides to this. I think it is better style to have it at the top of the file though.